gcc/rust/hir/tree/rust-hir-expr.h

```diff
@@ -1105,8 +1108,8 @@ class ArrayIndexExpr : public ExprWithoutBlock
   void accept_vis (HIRFullVisitor &vis) override;
   void accept_vis (HIRExpressionVisitor &vis) override;
 
-  Expr *get_array_expr () { return array_expr.get (); }
-  Expr *get_index_expr () { return index_expr.get (); }
+  std::unique_ptr<Expr> &get_array_expr () { return array_expr; }
```

Comment:
You are amazing this has been bugging me that the HIR is super inconsistent like that and returning raw pointers. Thanks for doing this

Reply:
I was not really sure what to do, but I've noticed the inconsistencies. I can fix them along the way, that's easy and helps me understand the HIR anyway :)